### ISSUE TITLE
New chip?

### ISSUE THREAD

**Issue by StomperQc** [RoboDurden/Hoverboard-Firmware-Hack-Gen2.x#55]
I bought some refirbushed hoverboard for cheap and it's a split board with a new chip i guess, i have a lot of those board so if you want one ill send a pair to you and ill pay for shipping, so you can test it.

Brand: Gyrocopter
36v battery

The chip brand is LKS and the first number on it is lks32mc

![20240125_093403](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/157717980/ebbd29f8-d2ac-4529-b38e-f1333b1c0db2)
![20240125_092151](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/157717980/f6aeb383-253a-4d0c-b487-801a685b8c37)
![20240125_091925](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/157717980/c28f9f6c-afc8-455a-907c-b532f2b0a9c5)
![20240125_092211](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/157717980/cd793d50-9c08-45dc-bff0-eb84d4f04771)

**Comment by AILIFE4798** [comment]
lks is back...

**Comment by RoboDurden** [comment]
Yes, i already search the other issue ...

**Comment by AILIFE4798** [comment]
if not for the mm32spin then i may be able to try support it

**Comment by AILIFE4798** [comment]
but lets be real the hoverboard looks very shtty the mosfet is tiny and plastic wheel really not worth support this
and no opamp on board (but lks32 have one built in the chip)

**Comment by RoboDurden** [comment]
This is the other issue: https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/issues/29 "Support for board YFS-N2? "

**Comment by RoboDurden** [comment]
@AILIFE4798 ,  maybe you again can find the dataship for this MCU. The other mcu seems to not only have a dual opamp but also a 6ch gate driver inside: LKS32MC03x with built-in 6N Gate Driver

I have so many good old hoverboard motors, that i do not care so much for these newer plastic motors. And as i said, there are lots of applications like lawn mover, where the build quality of the motors is not cruicial.

And smd mosefts does not need to be worse than TO220 ones. And again, there are many use cases where 350+ Watt are not needed.

I fear that these LKS mcu are not pin compatible to the GD32.

So i think we may try LKS only after we have mastered the MM32 mcu..

**Comment by AILIFE4798** [comment]
can u give me full marking on chip i cannot read it

**Comment by AILIFE4798** [comment]
https://datasheet.lcsc.com/lcsc/2110081130_LINKOSEMI-LKS32MC088C6T8_C2798416.pdf

**Comment by AILIFE4798** [comment]
phone is a very good tool to trace a board manually. most phone have a macro lens now,with flashlight on it can deliver a very good quality photo,and also good if you need to solder 2 wire to the mcu pin...

**Comment by AILIFE4798** [comment]
https://www.lksmcu.com/index.php/DevicePack-en/
keil pack

**Comment by AILIFE4798** [comment]
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/142502122/f6bbcdcd-7df1-46c2-996a-aa10d2f1fb45)
pinout is not compatiable

**Comment by RoboDurden** [comment]
With dedicated hall and mosfet pins, this will make tracing such a board easier..
@StomperQc , for now it is up to you to install Keil and import the device pack that @AILIFE4798 has linked to.
When you then succeed to successfully compile a blink example,
i will be happy to add a LKS32 target to this firmware.

Thanks for the offer to ship such a hoverboard to me (Germany), but i still have many other Gen2 boards here i first want to trace with my autodetect this year.
Where are you from ? May shipping to @AILIFE4798 would also be an option :-)
He by now is more experienced in adding new targets to the firmware !

**Comment by RoboDurden** [comment]
Again, 32kB are absolutly okay for this Gen2.x firmware.
And when you have succeeded with MM32, i might very well like to port the FOC example code that will fit into the 32kB.

**Comment by AILIFE4798** [comment]
lks32 is also capable of foc and it have dsp to calculate the foc math so the mcu dont have to
the comparator/opamp is also in the mcu and so itll make the board even cheaper but also harder to code
it is really dedicated so it will not be easy if the example code did not work like the mm32

**Comment by AILIFE4798** [comment]
you can send one pair to me but i may have to work on mm32 so itll get delayed a bit

**Comment by StomperQc** [comment]
Im from Canada, i dont mind sending a pair to @AILIFE4798
As i said i have many of them, i wont be able to do it by myself i dont have the tech knowledge so i dont mind shipping a pair if i and others can reuse them later instead of trowing them out.

**Comment by AILIFE4798** [comment]
you can send a pair i will try to support it when i have time
im from china so it wouldnt be very easy nor cheap to send a full hoverboard so i think ill have to find another one to selvage the wheel which is a bit annoying but at least i can tell you how to upload the code to it
this mcu is very new so there is limited resources for programming it but there is example
i will send you my address if you want to
or you try it your self and ill provide all the resources available from random chinese website

**Comment by StomperQc** [comment]
Yes im willing to send a pair of board to china, it would cost around 20-30$ so it's not that bad.

You can send me your adress at klempstock@hotmail.com and ill let you know when it shipped.

I dont mind if it takes time i understand everybody is doing it on their own free time.

**Comment by AILIFE4798** [comment]
one of the board is 5v and the other is 3.3...

![lks voltage](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/142502122/b5eb20f8-6562-4e67-a03d-3e9dd3f4ec95)

**Comment by AILIFE4798** [comment]
lks32 also have the example foc program(17kb) and compile without error after minor modification
so it wouldnt be too hard to support just like the mm32
but i think the mm32 example is simpler to implement so ill use that

**Comment by I-hate-2FA** [comment]
There is keil device pack for it on the official website 
If you want to make auto detect you can port gd32or mm32 auto detect, but if all of your board is same tracing manually can be done in an hour
